src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -314,4 +330,53 @@ public void captures_correctly_when_captor_used_on_pure_vararg_method() throws E
         verify(mock).varargs(eq(42), argumentCaptor.capture());
         Assertions.assertThat(argumentCaptor.getValue()).contains("capturedValue");
     }
+
+    @Test
+    public void should_capture_only_list_instances() throws Exception {
+        //given
+        ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+
+        //when
+        mock.forCollection(new ArrayList<String>());
+        mock.forCollection(new HashSet<String>());
+        mock.forCollection(new LinkedList<String>());
+
+        //then
+        verify(mock, times(2)).forCollection(captor.captureIf(new InstanceOf(List.class)));
+        assertThat(2, is(equalTo(captor.getAllValues().size())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_equal_to_a_certain_value() throws Exception {
+        //given
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+
+        //then
+        verify(mock).simpleMethod(captor.captureIf(new Equals(FOO)), anyInt());
+        assertThat(1, is(equalTo(captor.getAllValues().size())));
+        assertThat(FOO, is(equalTo(captor.getValue())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_less_then() {
+        //given
+        ArgumentCaptor<Integer> captor = ArgumentCaptor.forClass(Integer.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+        mock.simpleMethod(FOO, 3);
+        mock.simpleMethod(BAR, 4);
+        
+        mock.simpleMethod(FOO, 5);
+
+        //then
+        verify(mock, times(2)).simpleMethod(anyString(), captor.captureIf(new LessThan<Integer>(3)));
```

Comment:
I tried to implement as much as possible. Still, I have issues's in implementing the `CapturingMatcher` using `AdditionalMatchers`.
What do you think about replacing `captureIf` with `captureIfIsA`, and so on?

An example would be:
```java
Processor mock = ...

mock.process(new ArrayList<>);
mock.process(new HashSet<>);

ArgumentCaptor captor = ...

verify(mock).process(captor.captureIfIs(ArrayList()));
```

and the implementation of this would be

```java
public T captureIfIsA(Class clazz) {
    new CapturingMatcher(new InstanceOf(clazz), arguments);
...
}
```

Thanks for your ideas.